lightkurve/mast.py

```diff
@@ -61,86 +69,242 @@ def search_kepler_products(target):
+        mask &= np.array([desc.lower().endswith('fits') or
+                          desc.lower().endswith('fits.gz')
+                          for desc in products['dataURI']])
+        mask &= np.array([suffix in desc for desc in products['description']])
+        products = products[mask]
 
+    #Add the quarter or campaign numbers
+    qoc = np.asarray([p.split(' - ')[-1][1:]
+                      for p in products['description']], dtype=int)
+    products['qoc'] = qoc
+     #Add the dates of each short cadence observation to the product table.
+    dates = [p.split('/')[-1].split('-')[1].split('_')[0]
+             for p in products['dataURI']]
+    for idx, d in enumerate(dates):
+        try:
+            dates[idx] = float(d)
+        except:
+            dates[idx] = 0
+    products['dates'] = np.asarray(dates)
+    return products
 
-def search_kepler_lightcurve_products(target, cadence='long', quarter=None,
-                                      campaign=None):
-    """Returns a table of Kepler or K2 lightcurve files for a given target.
+def get_kepler_products(target, filetype='Target Pixel', cadence='long',
+                        quarter=None, month=None, campaign=None, radius=1.,
+                        targetlimit=1, verbose=True, **kwargs):
+    """Fetch files from from the Kepler/K2 data archive at MAST.
 
-    This only returns products produced by the official Kepler pipeline,
-    which is not necessarily the best choice for every use case.
+    Raises an `ArchiveError` if a unique TPF cannot be found.  For example,
+    this is the case if a target was observed in multiple Quarters and the
+    quarter parameter is unspecified.
 
     Parameters
     ----------
-    cadence: 'short' or 'long'
-        Specify short (1-min) or long (30-min) cadence data.
-    quarter, campaign : int
-        Specify the Kepler Quarter or K2 Campaign Number.
-        If None, then return the products for all Quarters/Campaigns.
+    target : str or int
+        KIC/EPIC ID or object name.
+
+    filetype : 'Target Pixel' or 'Lightcurve'
+        Whether to return TPFs of LCs
+    cadence : str
+        'long' or 'short'.
+    quarter, campaign : int, list or 'all'
+        Kepler Quarter or K2 Campaign number.
+    month : 1, 2, 3, list or 'all'
+        For Kepler's prime mission, there are three short-cadence
+        Target Pixel Files for each quarter, each covering one month.
+        Hence, if cadence='short' you need to specify month=1, 2, or 3.
+    radius : float
+        Search radius in arcseconds. Default is 1 arcsecond.
+    targetlimit : None or int
+        Limit the number of returned target pixel files. If none, no limit
+        is set
+    kwargs : dict
+        Keywords arguments passed to `KeplerTargetPixelFile`.
 
     Returns
     -------
-    products : astropy.Table
-        Table detailing the available Target Pixel File products.
+    path : str or list of strs
```

Comment:
The return value is `return np.asarray(path)` so I presume it always returns an `array` now, not a `str` or a `list`?

Reply:
Yeah it's always an array. A list is probably better.